### PR TITLE
Add snippet for weblinc adapter.js

### PR DESCRIPTION
**Pull request by JakeBeresford**
Added a new snippet to for adapter js files

### CODE REVIEW

**Review by mupt — COMMENTED** [review]
You may want to include a pattern for including current_user data in your callback.  That is a frequent requirement of these adapters.

'addToCart': function () {
var addToCartEvent = function (payload) {
}
WEBLINC.currentUser.gettingUserData.done(_.partial(addToCartEvent);
) 
}

File: weblinc-js-adapter.sublime-snippet [hunk 1]
```diff
@@ -0,0 +1,109 @@
...
+            }
+
+            ${2://add your send logic here}
+        },
+
+        isLoaded = function () {
+            var scripts = _.toArray(document.scripts),
+
+            loaded = _.any(scripts, function (script) {
+                return script.src.match(/someregexthatmatchesyour3rdpartyscript/i);
```
Comment (meowsus) [inline weblinc-js-adapter.sublime-snippet]: `someregexthatmatchesyour3rdpartyscript` should be `${3:someregexthatmatchesyour3rdpartyscript}`

Comment (meowsus) [reply weblinc-js-adapter.sublime-snippet]: obvs you'll have to increment the other `${n:...}` accordingly

File: weblinc-js-adapter.sublime-snippet [hunk 2]
```diff
@@ -0,0 +1,109 @@
...
+                if (WEBLINC.analytics.debug) {
+                    console.log('WEBLINC.analytics: trying to send page view again');
+                }
+            }
+        };
+
+    // This is a good place to add any global objects for your integration, like dataLayers
+
+    return {
+        ${4:// Base callbacks - remove any that you do not need}
```
Comment (meowsus) [inline weblinc-js-adapter.sublime-snippet]: I think this is superfluous.

File: weblinc-js-adapter.sublime-snippet [hunk 3]
```diff
@@ -0,0 +1,109 @@
...
+        'signup': function () {},
+
+        'share': function (payload) {},
+
+        'emailSignup': function () {}
+    };
+});
+]]></content>
+  <!-- Optional: Set a tabTrigger to define how to trigger the snippet -->
+  <tabTrigger>wl-adapter</tabTrigger>
```
Comment (meowsus) [inline weblinc-js-adapter.sublime-snippet]: Let's change this to `wl-js-analytics-adapter`

I'm about to namespace the others in this branch.

**Review by meowsus — CHANGES_REQUESTED** [review]

**Comment by meowsus** [comment]
Also, I like FM's idea.

File: wl-js-analytics-adapter.sublime-snippet [hunk 4]
```diff
@@ -0,0 +1,113 @@
...
+            }
+
+            ${2://add your send logic here}
+        },
+
+        isLoaded = function () {
+            var scripts = _.toArray(document.scripts),
+
+            loaded = _.any(scripts, function (script) {
+                return script.src.match(WEBLINC.config.${1:adapterName}.scriptRegex);
```
Comment (meowsus) [inline wl-js-analytics-adapter.sublime-snippet]: If this is 1-to-1, meaning that this adapter is a client for only one third party, i think that adding config is an additional step that isn't needed. 

Thoughts?

Comment (meowsus) [reply wl-js-analytics-adapter.sublime-snippet]: Also, any thoughts around FMs idea to add support for the currentUser promise?

Comment (JakeBeresford) [reply wl-js-analytics-adapter.sublime-snippet]: I guess it depends what the adapter is doing, and what information is required for the integration. My thought here was that there may be API keys, APP ID etc. I used a config for the personalized messaging / free shipping thing and thought it was a good idea. Happy to remove it if that's the consensus.

I'd like to include support for currentUser, I'm will check out how others have implemented it to find the nicest solution. Is there a 'best practice' way to access currentUser?

File: wl-js-analytics-adapter.sublime-snippet [hunk 5]
```diff
@@ -20,14 +20,9 @@ WEBLINC.analytics.registerAdapter('${1:adapterName}', function () {
             var scripts = _.toArray(document.scripts),
 
             loaded = _.any(scripts, function (script) {
-                return script.src.match(WEBLINC.config.${1:adapterName}.scriptRegex);
+                return script.src.match(/{3:someregexthatmatchesyour3rdpartyscript}/i);
```
Comment (meowsus) [inline wl-js-analytics-adapter.sublime-snippet]: You didn't increment the numbers that follow. Line 38 (for example) also is a "3".

Comment (meowsus) [inline wl-js-analytics-adapter.sublime-snippet]: Should be proceeded by a `$`

**Review by meowsus — CHANGES_REQUESTED** [review]

**Pull request merged**